src/XRP/default-xrp-client.ts

```diff
@@ -495,6 +496,43 @@ export default class DefaultXrpClient implements XrpClientDecorator {
     return await this.getTransactionResult(transactionHash)
   }
 
+  /**
+   * Disables DepositPreauth and unauthorizes an XRPL account to deposit to this XRPL account.
+   *
+   * @set https://xrpl.org/depositpreauth.html
+   *
+   * @param xAddressToUnauthorize The X-Address of the account to unauthorize DepositPreauth for.
+   * @param wallet The wallet associated with the XRPL account disabling DepositPreauth and that will sign the request.
+   */
+  public async unauthorizeDepositPreauth(
+    xAddressToUnauthorize: string,
+    wallet: Wallet,
+  ): Promise<TransactionResult> {
+    const classicAddress = XrpUtils.decodeXAddress(xAddressToUnauthorize)
+    if (!classicAddress) {
+      throw XrpError.xAddressRequired
+    }
+
+    const senderAccountAddress = new AccountAddress()
+    senderAccountAddress.setAddress(xAddressToUnauthorize)
+
+    const unauthorize = new Unauthorize()
+    unauthorize.setValue(senderAccountAddress)
+
+    const depositPreauth = new DepositPreauth()
+    depositPreauth.setUnauthorize(unauthorize)
+
+    const transaction = await this.prepareBaseTransaction(wallet)
+    transaction.setDepositPreauth(depositPreauth)
+
+    const transactionHash = await this.signAndSubmitTransaction(
+      transaction,
+      wallet,
+    )
+
+    return await this.getTransactionResult(transactionHash)
+  }
```

Comment:
i chose to copy+paste this implementation instead of being more clever. i tried some other clever approaches (namely, having a bool to switch behavior), but it didn't feel right to me. authorize/unauthorize do have slightly different behavior so i decided to wait. if you feel strongly about his, let me know and i can try to reuse the logic